### ISSUE TITLE
Gateway will use the default rpc client

### ISSUE THREAD

**Issue by douglaz** [fedimint/fedimint#2955]
It seems the gateway will try to use the bitcoind defined on the wallet module consensus configuration, like:
```json
    "2": {
      "kind": "wallet",
      "client_default_bitcoin_rpc": {
        "kind": "esplora",
        "url": "http://127.0.0.1:50002/"
      },
```
Ignoring all local configuration like FM_BITCOIN_* variables

(Perhaps this problem isn't gateway specific and will happen with any client?)

**Comment by elsirion** [comment]
Yes, the client doesn't take env vars into account. The problem is that there is a wide variety of clients and only a subset of them supports env vars, so it didn't occur to me.

I think it would be interesting to support as long as there is no better way of overriding this default (which will likely be stateful and more complex).

**Comment by justinmoon** [comment]
Gatewayd could read from env [here](https://github.com/fedimint/fedimint/blob/2bb733ad6ac818f89cb2f00f352c0ce9896997ed/gateway/ln-gateway/src/lib.rs#L185-L190) and set the client bitcoin rpc inside `WalletClientGen` instead of just using `WalletClientGen::default()`. This seems cleaner than having client library ready from environment.

**Comment by m1sterc001guy** [comment]
This seems like a configuration that would be good to add to the gateway's config API: https://github.com/fedimint/fedimint/pull/3154

I think most gateway operators will only want to configure their bitcoind instance once, and not per-client.

**Comment by m1sterc001guy** [comment]
This is blocked on updating `bitcoincoire-rpc` to 0.17.0. The client can't currently use bitcoind for RPCs because it calls `import_address_script`, which throws an error when called on descriptor wallets. I believe this needs to be upgraded to `import_descriptors` which is available in `bitcoincore-rpc` 0.17.0

**Comment by elsirion** [comment]
https://github.com/fedimint/fedimint/issues/7235 should address this too.
